maitake-sync/src/wait_cell.rs

```diff
@@ -327,6 +327,46 @@ impl WaitCell {
         }
     }
 
+    /// Asynchronously poll the [`WaitCell`] until a condition occurs
+    ///
+    /// This can be used to implement a "wait loop", turning a "try" function
+    /// (e.g. "try_recv" or "try_send") into an asynchronous function (e.g.
+    /// "recv" or "send").
```

Comment:
another use case is waiting for an interrupt handler to set a bit or something --- there's actually a lot of code in some of the mnemOS drivers that are basically manual implementations of this pattern, e.g. https://github.com/tosc-rs/mnemos/blob/45457b0e72d1bfb755d700f982964092fc116512/platforms/esp32c3-buddy/src/drivers/usb_serial.rs#L96-L104. it might be good to mention something like that here?